fighters/common/src/general_statuses/attackdash.rs

```diff
@@ -131,26 +131,102 @@ unsafe extern "C" fn sub_attack_dash_uniq(fighter: &mut L2CFighterCommon, arg: L
 
 #[hook(module = "common", symbol = "_ZN7lua2cpp16L2CFighterCommon22status_AttackDash_MainEv")]
 unsafe extern "C" fn status_AttackDash_Main(fighter: &mut L2CFighterCommon) -> L2CValue {
-    if CancelModule::is_enable_cancel(fighter.module_accessor) && fighter.sub_wait_ground_check_common(L2CValue::Bool(false)).get_bool() {
+    if CancelModule::is_enable_cancel(fighter.module_accessor)
+    && fighter.sub_wait_ground_check_common(L2CValue::Bool(false)).get_bool()
+    || fighter.sub_air_check_fall_common().get_bool() {
         // Clear motion energy once dash attack is interrupted
         // to prevent slipping off edge during buffered action
-        fighter.clear_lua_stack();
-        lua_args!(fighter, FIGHTER_KINETIC_ENERGY_ID_MOTION);
-        app::sv_kinetic_energy::clear_speed(fighter.lua_state_agent);
+        sv_kinetic_energy!(
+            clear_speed,
+            fighter,
+            FIGHTER_KINETIC_ENERGY_ID_MOTION
+        );
         return L2CValue::I32(0);
     }
-    /*if fighter.global_table[SITUATION_KIND] == SITUATION_KIND_AIR && !VarModule::is_flag(fighter.battle_object, vars::common::ATTACK_DASH_SLIDEOFF) {
-        if ParamModule::get_flag(fighter.module_accessor, ParamType::Shared, "attack_dash_cliff_stop") {
-            fighter.change_status(
-                L2CValue::I32(*FIGHTER_STATUS_KIND_FALL),
-                L2CValue::Bool(false)
-            );
-            return L2CValue::I32(0);
-        } else {
-            KineticModule::change_kinetic(fighter.module_accessor, *FIGHTER_KINETIC_TYPE_FALL);
-            VarModule::on_flag(fighter.battle_object, vars::common::ATTACK_DASH_SLIDEOFF);
+    // <HDR>
+    let situation = fighter.global_table[SITUATION_KIND].get_i32();
+    // This block is to force a ground correct type depending on if you enable sliding off or not.
+    if VarModule::is_flag(fighter.battle_object, vars::common::status::ATTACK_DASH_ENABLE_AIR_FALL)
+    && situation == *SITUATION_KIND_GROUND {
+        GroundModule::correct(fighter.module_accessor, GroundCorrectKind(*GROUND_CORRECT_KIND_GROUND));
+    }
+    // This block checks if you want to enable air drift. This code will only run once, and only while in the air,
+    // but it enables another flag that will be checked when your situation changes and renable the right kinetic type there.
+    if VarModule::is_flag(fighter.battle_object, vars::common::status::ATTACK_DASH_ENABLE_AIR_DRIFT)
+    && situation != *SITUATION_KIND_GROUND {
+        VarModule::off_flag(fighter.battle_object, vars::common::status::ATTACK_DASH_ENABLE_AIR_DRIFT);
+        VarModule::on_flag(fighter.battle_object, vars::common::status::ATTACK_DASH_AIR_DRIFT_ENABLED);
+        KineticModule::change_kinetic(fighter.module_accessor, *FIGHTER_KINETIC_TYPE_FALL);
```

Comment:
it wouldn't work, because the last time Control is enabled is during DASH *because it's required to do dash attack), simply enabling control here would use the kinetic params for DASH. Or at least, that's my understanding of it anyway.
changing kinetic type to KINETICC_TYPE_FALL here is much simpler as it carries your MOTION momentum *and* uses the right params for air drift.